oras/auth/__init__.py

```diff
@@ -14,10 +14,11 @@ class AuthenticationException(Exception):
     pass
 
 
-def get_auth_backend(name="token", session=None, **kwargs):
+def get_auth_backend(name="token", session=None, prefix="https", **kwargs):
     backend = auth_backends.get(name)
     if not backend:
         raise ValueError(f"Authentication backend {backend} is not known.")
     backend = backend(**kwargs)
     backend.session = session or requests.Session()
+    backend.prefix = prefix
```

Comment:
step 3.
the determined backend (either token or basic) get `.prefix` property set to the value received in step 2

this maintains code-style for `.session` another property set on determined backend.